crates/trippy-core/src/tos.rs

```diff
@@ -0,0 +1,206 @@
+use crate::{Error, TypeOfService};
+use std::fmt::Display;
+
+/// Explicit Congestion Notification (`ECN`).
+///
+/// This is used in the `ECN` field of the `IP` header.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum ExplicitCongestionNotification {
+    /// Not ECN-Capable Transport (00, 0 dec)
+    NotECT,
+    /// ECN Capable Transport(1) (01, 1 dec)
+    ECT1,
+    /// ECN Capable Transport(0) (10, 2 dec)
+    ECT0,
+    /// Congestion Experienced (11, 3 dec)
+    CE,
+}
+
+impl TryFrom<u8> for ExplicitCongestionNotification {
+    type Error = Error;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::NotECT),
+            1 => Ok(Self::ECT1),
+            2 => Ok(Self::ECT0),
+            3 => Ok(Self::CE),
+            _ => Err(Error::Other(format!("invalid ECN value: {value}"))),
+        }
+    }
+}
+
+impl Display for ExplicitCongestionNotification {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::NotECT => write!(f, "Not ECN-Capable Transport"),
+            Self::ECT1 => write!(f, "ECN Capable Transport(1)"),
+            Self::ECT0 => write!(f, "ECN Capable Transport(0)"),
+            Self::CE => write!(f, "Congestion Experienced"),
+        }
+    }
+}
+
+/// Differentiated Services Code Point (`DSCP`).
+///
+/// This is used in the `DSCP` field of the `IP` header.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum DifferentiatedServicesCodePoint {
+    /// Best Effort (000 000, 0 dec)
+    BE,
```

Comment:
I know I called this best effort, but going back to rfc2474/2475 the name would be a clunky Default PHB, this was later streamlined into Default Forwarding or DF for short.
This is also one of the class-selector codepoints, specifically CS0 (rfc2474)

Reply:
I changed it to be `DF`.